.github/workflows/docker.yml

```
@@ -0,0 +1,51 @@
+---
+name: Build and push Docker image
+on:
+  push:
+    branches:
+      - main
+      - add/docker_ci
+    paths:
+      - Dockerfile
+      - conda-lock.yml
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Maximize build space
+        uses: easimon/maximize-build-space@master
+        with:
+          remove-dotnet: "true"
+          remove-codeql: "true"
+          remove-haskell: "true"
+          remove-android: "true"
+          overprovision-lvm: "true"
+          remove-docker-images: "true"
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: amd64
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          registry: ${{secrets.DOCKER_REGISTRY}}
+          username: ${{secrets.DOCKER_REGISTRY_USERNAME}}
+          password: ${{secrets.DOCKER_REGISTRY_TOKEN}}
+      - name: Get torch version for amd
+        run: "echo TORCH_VERSION_AMD=$(cat conda-lock.yml | grep -E -A 4 'name: pytorch$' |
+          tail -n1 | grep -Eo [0-9]+.[0-9]+.[0-9]+) >> $GITHUB_ENV"
+      - name: Get cuda version for amd
+        run: "echo CUDA_VERSION_AMD=$(cat conda-lock.yml | grep -E -A 4 'name:
+          pytorch-cuda$' | tail -n1 | grep -Eo [0-9]+.[0-9]+) >> $GITHUB_ENV"
+      - name: Build and push AMD
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          platforms: linux/amd64
+          push: true
+          tags: ${{secrets.DOCKER_REPOSITORY}}:pytorch-${{env.TORCH_VERSION_AMD}}-cuda-${{env.CUDA_VERSION_AMD}}, ${{secrets.DOCKER_REPOSITORY}}:latest
```

```suggestion
          tags: konstantinjdobler/nlp-research-template:pytorch${{env.TORCH_VERSION_AMD}}-cuda${{env.CUDA_VERSION_AMD}}, konstantinjdobler/nlp-research-template:latest
```

We can just hardcode it since it's public anyways, right? Makes it easier to change when using the template

+ use tag version scheme from PyTorch